app/dao/events_dao.py

```diff
@@ -1,6 +1,22 @@
 from app import db
+from app.models import Event
 
 
 def dao_create_event(event):
-    db.session.add(event)
+    """
+    Given a dictionary of event data like . . .
+
+    {'event_type': 'sucessful_login', 'data': {'in_fact': 'could be anything', 'something': 'random'}}
+
+    . . . persist a new Event instance.
+    """
+
+    if isinstance(event, dict):
+        event_instance = Event(**event)
+    elif isinstance(event, Event):
+        event_instance = event
+    else:
+        raise TypeError(f"Event is of type {type(event)}.")
+
+    db.session.add(event_instance)
```

Comment:
These changes, and similar changes in other files, are a quick fix.  After the upgrades, the db.session.add call was failing when the input was a dictionary rather than an instance.  This problem might actually be fixed independently by using the "load_instance" parameter in the "meta" class of the various schema definitions in app/schemas.py.